distributed/tests/test_steal.py

```diff
@@ -145,23 +146,51 @@ def test_steal_related_tasks(e, s, a, b, c):
 
 
 @gen_cluster(client=True, nthreads=[("127.0.0.1", 1)] * 10, timeout=1000)
-def test_dont_steal_fast_tasks(c, s, *workers):
+async def test_dont_steal_fast_tasks_compute_time(c, s, *workers):
     np = pytest.importorskip("numpy")
     x = c.submit(np.random.random, 10000000, workers=workers[0].address)
 
     def do_nothing(x, y=None):
         pass
 
-    yield wait(c.submit(do_nothing, 1))
+    # execute and meassure runtime once
+    await wait(c.submit(do_nothing, 1))
 
     futures = c.map(do_nothing, range(1000), y=x)
 
-    yield wait(futures)
+    await wait(futures)
 
     assert len(s.who_has[x.key]) == 1
     assert len(s.has_what[workers[0].address]) == 1001
 
 
+@gen_cluster(client=True, nthreads=[("127.0.0.1", 1)] * 10, timeout=1000)
+async def test_dont_steal_fast_tasks_blacklist(c, s, *workers):
+    import time
+
+    # create a dependency
+    x = c.submit(slowinc, 1)
+
+    def fast_blacklisted(x, y=None):
+        time.sleep(0.01)
+        pass
+
+    mmock = mock.MagicMock()
+    s.extensions["stealing"].move_task_request = mmock
+    import distributed.stealing
+
+    with mock.patch.object(
+        distributed.stealing, "fast_tasks", {fast_blacklisted.__name__}
+    ):
+
+        w = workers[0]
+        futures = c.map(fast_blacklisted, range(100), y=x)
+
+        await wait(futures)
+
+    mmock.assert_not_called()
```

Comment:
This test is highly dependent on the internal API.  Historically we've tried to avoid tests like this, because it makes changing internal code really hard (future developers have to understand and change all tests that depend on the internal API).  Instead, I wonder if you could use something like `c.map(..., workers=[...], allow_other_workers=True)` and then test that none of the other workers got any data.  This only requires that we keep the `worker.data` API, which I think is probably more stable than the `move_task_request` API.

Also, do we need the `timeout=1000` here, or was this just for debugging?

Reply:
Thanks, I already suspected this to become an issue during review :)

I didn't know about the `allow_other_workers` flag, will rewrite the test, ofc.

- [ ] No mocks

> Also, do we need the timeout=1000 here, or was this just for debugging?

Yes, that was debugging. No timeout should be necessary here.

- [ ] No timeouts